compute_endpoint/globus_compute_endpoint/endpoint/interchange.py

```diff
@@ -370,6 +370,14 @@ def process_pending_tasks():
             # gracefully, iterate once a second whether a task has arrived.
             nonlocal num_tasks_forwarded
             while not self._quiesce_event.is_set():
+                if executor.bad_state_is_set:
+                    try:
+                        raise executor.executor_exception
+                    except Exception:
+                        log.exception("Engine has failed with an unrecoverrable error")
+                    finally:
+                        self.time_to_quit = True
+
```

Comment:
No. Both these are set by the same method here-> https://github.com/Parsl/parsl/blob/76b4d9d75673e518b48b2cdf0f364c3f8010cf8c/parsl/executors/status_handling.py#L135

Reply:
Right, that makes sense, I can just check `executor_exception ` directly. 

We want the endpoint to terminate as soon as the executor is failed and not necessarily wait for a task to come in to trigger the failure. We sort of need the endpoint to fail and disconnect so that the endpoint shows up as dead, until we add extra reporting of failure + reason upon endpoint shutdown.

Reply:
Ah, I see.  As in, if not tasks come in, we'd be idling for some time.  Okay.

I'd still rather we didn't raise an exception out of the thread, since we don't otherwise handle that.  Instead, please log, set time to quit and continue.  That leads to a single-point-of-exit vis-a-vis the `while` loop conditional.  So something like this is what's currently in my head given the context of which I'm aware:

```python
while not self.time_to_quit:
    if executor.executor_exception:
        log.exception(executor.executor_exception)  # if we need the traceback
        # otherwise log.error, warning, info, etc.
        self.stop()  # this sets self.time_to_quit and starts the quiesce
        continue
```

Reply:
I've updated this to 
```python
                if executor.executor_exception:
                    self.time_to_quit = True
                    log.exception(executor.executor_exception)
```

I'm setting `time_to_quite` because we call stop right below this.